### ISSUE TITLE
regression: sharp drop in performance under a certain circumstances

### ISSUE THREAD

**Issue by luobotaxinghu** [Microsoft/LightGBM#1298]
In regression,  when  standard deviation of  target value is more than 10,000,  there is  a sharp drop in performance. 
The code  below,  show that origin  y in [-500, 500] mostly,  multiply by 1000 and y is in [-500k, 500k]. You will find origin predict value of lightgbm is in [-500, 500],  but it is in [-50k, 50k] after multiplying. 
I would like to know why and how to deal with it.

code: 

`
from lightgbm import LGBMRegressor
from sklearn.datasets import make_regression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

x, y = make_regression(n_samples=5000, n_features=10, random_state=2018)
y *= 1000

X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.1, random_state=2018)

params = {
    'max_depth': 50,
    'task': 'train',
    'boosting_type': 'gbdt',
    'objective': 'regression',
    'metric': {'l2'},
    'num_leaves': 20,
    'min_data_in_leaf': 50,
    'learning_rate': 0.2,
    "n_estimators": 100,
}
lgb_model = LGBMRegressor(**params)

lgb_model.fit(X=X_train, y=y_train)
lgb_pred = lgb_model.predict(X=X_test)
mse = mean_squared_error(y_test, lgb_pred)
print(np.sqrt(mse))

real = y_test.reshape(-1, )
result = pd.DataFrame(np.stack([lgb_pred, real], axis=1), columns=["pred", "real"])
result.plot()
plt.show()
`

origin y: 
![image](https://user-images.githubusercontent.com/16847652/38245490-d529c022-3770-11e8-8c03-6bed48d08165.png)
y *1000:
![image](https://user-images.githubusercontent.com/16847652/38245510-ec7ffe58-3770-11e8-842d-077daab1304a.png)

**Comment by kylelee82** [comment]
This may be related to the issue I had: https://github.com/Microsoft/LightGBM/issues/1278 (I believe it's the large regression numbers that cause a problem, rather than the standard deviation of the target)

**Comment by luobotaxinghu** [comment]
@kylelee82  Not because of  large regression numbers, you can add 1 million to y,  the performance will not drop. In a addition, XGB will  perform much better  with  similar  hyper-parameters.

**Comment by guolinke** [comment]
@luobotaxinghu `num_leaves=20` with `max_depth=50` ? 

It is because by large regression target. LightGBM will limit the max tree output to 100 (https://github.com/Microsoft/LightGBM/blob/master/include/LightGBM/tree.h#L14), as a result, for large regression target, it needs much more trees to convergence. You can change `100` to larger number or set `reg_sqrt=True`. However, large regression target is always not recommended. 

LightGBM will start the training from the average label, as a result, it still can learn well when you add 1 million to the labels.

**Comment by luobotaxinghu** [comment]
@guolinke Thank you.
